tools/link.py

```diff
@@ -2098,34 +2089,20 @@ def phase_source_transforms(options):
 # both main code and libraries.
 # See also: `preprocess` in parseTools.js.
 def fix_es6_import_statements(js_file):
-  if not settings.MODULARIZE:
+  if not settings.MODULARIZE and not settings.AUDIO_WORKLET:
     return
-
   src = read_file(js_file)
-  write_file(js_file, src
-             .replace('EMSCRIPTEN$IMPORT$META', 'import.meta')
-             .replace('EMSCRIPTEN$AWAIT$IMPORT', 'await import')
-             .replace('EMSCRIPTEN$AWAIT(createWasm())', 'await createWasm()')
-             .replace('EMSCRIPTEN$AWAIT(', 'await ('))
+  if settings.MODULARIZE:
+    src = (src.replace('EMSCRIPTEN$IMPORT$META', 'import.meta')
+              .replace('EMSCRIPTEN$AWAIT$IMPORT', 'await import')
+              .replace('EMSCRIPTEN$AWAIT(createWasm())', 'await createWasm()')
```

Comment:
Wait not related to this PR as I see it was added earlier, but why are these necessary, does Closure not support `await` at all? IIRC, the replacements above were necessary specifically for unsupported `import.meta` and dynamic `import()`, but in other contexts it should work as-is.

Reply:
The `await createWasm()` is inside the closure / factory function when running in MODULARIZE mode.

However because closure is only run on the inner code it think its looking at a top-level await and it errors out.  At least that is my memory of why this is needed.

Running closure on the full/final output would be better, but I failed to land that change last time I tried because it caused a size regression (closure wan't able to optimize quite a a well for some reason).